### DISCUSSION TITLE
problems about './chainsaw search -t  'Event.System.EventID: =4104''

### DISCUSSION THREAD

**Discussion by cccrrrzz** [WithSecureLabs/chainsaw#123] [Q&A]
why some .evtx document error 'Failed to search file... - Failed to parse chunk number 7' or 10 or 2
 with code 
/chainsaw search -t  'Event.System.EventID: =4104'
![image](https://user-images.githubusercontent.com/128489507/226955062-f8639356-a658-4bd1-a389-0b48c85b8c40.png)

**Comment by alexkornitzer** [comment]
I assume this is the latest build of Chainsaw? If so I am going to probably need access to the log (if possible) that caused the error so I can see what the evtx library is not liking and get it reported/fixed upstream.

If you can't share it because its sensitive, if you run [evtx_dump](https://github.com/omerbenamram/evtx) and get the same error then we can raise the issue direct with upstream and work out how to proceed.

**Reply by cccrrrzz** [reply]
sorry bro, I am new githuber. can i ask how to share my (.evtx) document in github? or email it ?

**Reply by alexkornitzer** [reply]
If its not sensitive you should be able to drag and drop into message box.

**Reply by cccrrrzz** [reply]
[System.zip](https://github.com/WithSecureLabs/chainsaw/files/11042006/System.zip)

**Reply by cccrrrzz** [reply]
error like this
![image](https://user-images.githubusercontent.com/128489507/226964079-f2848de0-4d0c-49de-a2f8-3b57eae5527c.png)

**Reply by alexkornitzer** [reply]
Okay let me have a dig and i'll report back here. If it turns out to be a parsing issue i'll raise one from this. If the error is due to the file being corrupt (truncated or something like that) then I wont raise anything but the solution is to then use `--skip-errors` which will just breeze over the broken parts of the file.

**Reply by cccrrrzz** [reply]
thank u bro.

**Reply by alexkornitzer** [reply]
This looks like a truncated/corrupt file to me. If I load it in event view on windows I get the same number of events out as I do with the evtx library. So in thise cause you are safe to use `--skip-errors`

![Screenshot 2023-03-22 at 16 21 07](https://user-images.githubusercontent.com/2750747/226970694-b99aef31-a8e7-45a8-a715-c21f0d001a1a.png)
<img width="312" alt="Screenshot 2023-03-22 at 16 23 01" src="https://user-images.githubusercontent.com/2750747/226971224-bc57c906-30f5-46fb-ab5b-ed7d82338385.png">

Is that okay?

**Reply by cccrrrzz** [reply]
sorry bro for replying so late. (maybe we are not in the same TIME ZONE)
I add the code '--skip-errors' in the end of my code, OS shows nothing is found. Is it mean there is no record which 'ID=4104' in this file?
![image](https://user-images.githubusercontent.com/128489507/227118241-01c52256-f1ed-4fcb-a1a1-04209af41ba0.png)

**Reply by cccrrrzz** [reply]
And there is a question: Does adding the code '--skip-errors' lose some log info in report?
I guess this code '--skip-errors'   works on skipping  the error files, so if I want to make the all files' log report ,this report would miss the error logs' part

**Reply by cccrrrzz** [reply]
By the way, I am doing the windows system file log report, so the report must be complete and accurate

**Reply by alexkornitzer** [reply]
Yep so in this case the error is not a problem if you trust the source of the evtx, both Windows Event Viewer and the EVTX parser are both outputting the same number of events (1672). You are going to be getting no hits because there will be none in there with the ID 4104. If you do a different ID like 7036 you will get hits.

It should be outputting warnings if it skipped anything due to error, that looks like it might be a bug. I will have a look at that.

**Reply by cccrrrzz** [reply]
I took your suggestion and success ,thank u very much my bro

**Reply by hippyjm** [reply]
Good afternoon.
I cant seem to find some more explanation for why this error pops up when running Chainsaw.
I can see that in event viewer we have 2495 events and in evtx parser i get 2495 so the numbers match. However i dont fully understand why its throwing this error then. 
Everything is working a dream other than that. Its just having to check and make sure they match up


![Chainsaw error](https://github.com/WithSecureLabs/chainsaw/assets/137270373/577ad95e-7edb-4302-b3e5-ec828bf15975)
![EVTX parser](https://github.com/WithSecureLabs/chainsaw/assets/137270373/55f660c8-8496-450c-aab8-8bc1f9657e3b)

**Reply by alexkornitzer** [reply]
Hi @hippyjm, 

Without access to the sample its hard for me to say absolutely but either this is happening at the end of an evtx file as it could be truncated. And/or if the numbers are matching with Microsoft event viewer then usually its just the evtx library being more vocal about parsing errors.
